### PR TITLE
Refactored into library and MD bin.

### PR DESCRIPTION
**Pull request by frobnitzem**
I'm trying to create a Monte Carlo program from this, and have the following roadmap in-mind:

1. Break up cmake file into library and executable (to simplify work of adding multiple executables).
1. Create a test for energy calculations.
1. Build class/template-based input method for molecular topologies.
1. Create simplified route to calculation of Delta E-values.

This is step 1.  Not being a cmake expert, I don't know if `bin/CMakeLists.txt` is too verbose.  I think it doesn't need Kokkos::kokkos under under target_link_libraries, but maybe some of the others are redundant too because of cmake's linking rules.

### CODE REVIEW

**Comment by streeve** [comment]
Thanks for putting time into this. I created an issue that includes your main description to track progress - #33. We can update that as we go and make this PR more specific to step 1

I've also added a PR only for the CMake target changes #32 - if you merge that in here the build should pass.

**Comment by streeve** [comment]
@junghans any comments on CMake style?

File: CMakeLists.txt
Comment (junghans) [inline CMakeLists.txt]: Isn't it enough to look for Cabana?

Comment (streeve) [reply CMakeLists.txt]: Done in #32

**Comment by frobnitzem** [comment]
Accomplishes step 1 in Feature Request #33.  Merges suggestions from PR #32 (thanks @streeve).

`bin/CMakeLists.txt` is now particularly simple.  I don't know what you want to name the library and MD executable, but cmake only allows one target to be named `CabanaMD`.

File: CMakeLists.txt
Comment (junghans) [reply CMakeLists.txt]: ok

**Review by junghans — APPROVED** [review]

File: .travis.yml
Comment (junghans) [inline .travis.yml]: If you are using `CMAKE_CXX_COMPILER_LAUNCHER`, you can drop this line.

File: CMakeLists.txt
Comment (streeve) [inline CMakeLists.txt]: Looks like this got added back in the merge - can be removed

File: bin/CMakeLists.txt
Comment (streeve) [inline bin/CMakeLists.txt]: Maybe `cbnMD` instead?

File: .travis.yml
Comment (streeve) [inline .travis.yml]:
```suggestion
script:
  - export CXXFLAGS="-Wall -Wextra -pedantic -Werror"
```
Add to avoid my mistakes sticking around (thanks for fixing the one in read_data)

File: CMakeLists.txt
Comment (streeve) [inline CMakeLists.txt]: These can all be removed with cmakedefine, right?

File: CMakeLists.txt
Comment (streeve) [inline CMakeLists.txt]: I would prefer `unit_test` 

But I have some almost ready to push

File: src/CabanaMD_config.hpp.cmakein
Comment (streeve) [inline src/CabanaMD_config.hpp.cmakein:7]: Note that I've added one for Threads

**Review by streeve — CHANGES_REQUESTED** [review]
Just a few small changes. Probably easiest to wait until #32 is merged; there's quite a bit shifting with the Kokkos/Cabana updates

**Comment by streeve** [comment]
@frobnitzem should be fine to update - the CMake is even simpler now

File: .travis.yml
Comment (frobnitzem) [reply .travis.yml]: I removed the ln -s.

File: CMakeLists.txt
Comment (frobnitzem) [reply CMakeLists.txt]: This way, there's an error during configuration phase instead of during compile if the serial kokkos backend isn't enabled.

File: CMakeLists.txt
Comment (frobnitzem) [reply CMakeLists.txt]: changed to unit_test.

File: bin/CMakeLists.txt
Comment (frobnitzem) [reply bin/CMakeLists.txt]: adopted.

**Review by streeve — APPROVED** [review]
Looks good to me. Ready to merge, @frobnitzem?

**Comment by frobnitzem** [comment]
Yes, I think this one is ready.  Did I miss a checkbox somewhere?

**Comment by streeve** [comment]
Nope, just wanted to make sure

**Pull request merged**
